sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/GoneAndRetryWithRetryPolicy.java

```diff
@@ -360,7 +360,13 @@ public Mono<ShouldRetryResult> shouldRetry(Exception exception) {
                 Math.min(
                     Math.min(this.currentBackoffMilliseconds.get() + random.nextInt(RANDOM_SALT_IN_MS), remainingMilliseconds),
                     RetryWithRetryPolicy.MAXIMUM_BACKOFF_TIME_IN_MS));
-            this.currentBackoffMilliseconds.accumulateAndGet(RetryWithRetryPolicy.BACK_OFF_MULTIPLIER, (left, right) -> left * right);
+
+            this.currentBackoffMilliseconds.set(
+                Math.min(
```

Comment:
NIT - worth adding a Math.max(MINIMUM_BACKOFF_TIME_IN_MS, ...) as a safety net - I assume above is now safe for numeric overflows - but IMO would just make that assumption explicit

Reply:
added